pkg/distributor/distributor.go

```diff
@@ -415,12 +527,46 @@ func (d *Distributor) sendSamplesErr(ctx context.Context, ingester ring.Instance
 
 	_, err = c.(logproto.PusherClient).Push(ctx, req)
 	d.ingesterAppends.WithLabelValues(ingester.Addr).Inc()
-	if err != nil {
-		d.ingesterAppendFailures.WithLabelValues(ingester.Addr).Inc()
+
+	if err == nil {
+		return nil
+	}
+
+	d.ingesterAppendFailures.WithLabelValues(ingester.Addr).Inc()
+
+	// TODO: look for a better way to recognize a per-stream rate limit.
```

Comment:
I think this comment belongs more to the branch below, right? This confused me momentarily

Reply:
Oops, agreed. Moved it to the right place, thanks for catching it.